pkg/monitortests/clusterversionoperator/legacycvomonitortests/operators.go

```diff
@@ -56,18 +71,53 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
 			}
 
 			failures := testOperatorState(condition, operatorEvents, e2eEventIntervals)
-			if len(failures) > 0 {
-				ret = append(ret, &junitapi.JUnitTestCase{
-					Name:      testName,
-					Duration:  duration,
-					SystemOut: strings.Join(failures, "\n"),
-					FailureOutput: &junitapi.FailureOutput{
-						Output: fmt.Sprintf("%d unexpected clusteroperator state transitions during e2e test run \n\n%v", len(failures), strings.Join(failures, "\n")),
-					},
-				})
+			if len(failures) == 0 {
+				continue
+			}
+
+			excepted := []string{}
+			fatal := []string{}
+			for _, failure := range failures {
+				exception := ""
+				for regexp, e := range exceptions {
+					if regexp.MatchString(failure) {
+						exception = e
+						break
+					}
+				}
+				if exception == "" {
+					fatal = append(fatal, failure)
+				} else {
+					excepted = append(excepted, fmt.Sprintf("%s (exception: %s)", failure, exception))
+				}
+			}
+
+			output := fmt.Sprintf("%d unexpected clusteroperator state transitions during e2e test run", len(fatal))
+			if len(fatal) > 0 {
+				output = fmt.Sprintf("%s.  These did not match any known exceptions, so they cause this test-case to fail:\n\n%v\n", output, strings.Join(fatal, "\n"))
+			} else {
+				output = fmt.Sprintf("%s, as desired.", output)
+			}
+			output = fmt.Sprintf("%s\n%d unwelcome but acceptable clusteroperator state transitions during e2e test run", output, len(excepted))
+			if len(excepted) > 0 {
+				output = fmt.Sprintf("%s.  These should not happen, but because they are tied to exceptions, the fact that they did happen is not sufficient to cause this test-case to fail:\n\n%v\n", output, strings.Join(excepted, "\n"))
+			} else {
+				output = fmt.Sprintf("%s, as desired.", output)
+			}
+
+			ret = append(ret, &junitapi.JUnitTestCase{
+				Name:      testName,
+				Duration:  duration,
+				SystemOut: strings.Join(failures, "\n"),
+				FailureOutput: &junitapi.FailureOutput{
+					Output: output,
+				},
+			})
+
+			if len(fatal) == 0 {
+				// add a success so we flake and don't fail
+				ret = append(ret, &junitapi.JUnitTestCase{Name: testName})
```

Comment:
What do you think about returning one junit per *every* known cluster operator. This would allow us to get bugs to teams and track progress via the results of a specific test, otherwise we need to deciper generic test failures and parse output. Critically we'd need to return a successful junit for each when nothing is wrong otherwise test pass rates won't work.

I don't know of anywhere that we iterate all known clusteroperators, but scanning the intervals real quick for anything looking like this would make it pretty quick to build up a set:

```json
            "tempStructuredLocator": {
                "type": "ClusterOperator",
                "keys": {
                    "clusteroperator": "config-operator"
                }
            },
```

Reply:
Whoops just saw this is already there. :)